presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -362,37 +362,36 @@ public IPrestoSparkQueryExecution create(
                 authenticatorProviders);
 
         Session session = sessionSupplier.createSession(queryId, sessionContext);
-
         session = sessionPropertyDefaults.newSessionWithDefaultProperties(session, Optional.empty(), Optional.empty());
 
         WarningCollector warningCollector = warningCollectorFactory.create(getWarningHandlingLevel(session));
 
-        TransactionId transactionId = transactionManager.beginTransaction(true);
-        session = session.beginTransactionId(transactionId, transactionManager, accessControl);
-
-        // including queueing time
-        Duration queryMaxRunTime = getQueryMaxRunTime(session);
-        // excluding queueing time
-        Duration queryMaxExecutionTime = getQueryMaxExecutionTime(session);
-        // pick a smaller one as we are not tracking queueing for Presto on Spark
-        Duration queryTimeout = queryMaxRunTime.compareTo(queryMaxExecutionTime) < 0 ? queryMaxRunTime : queryMaxExecutionTime;
-
-        long queryCompletionDeadline = System.currentTimeMillis() + queryTimeout.toMillis();
-
-        queryMonitor.queryCreatedEvent(
-                new BasicQueryInfo(createQueryInfo(
-                        session,
-                        sql,
-                        PLANNING,
-                        Optional.empty(),
-                        sparkQueueName,
-                        Optional.empty(),
-                        queryStateTimer,
-                        Optional.empty(),
-                        warningCollector)));
-
         PlanAndMore planAndMore = null;
         try {
+            queryMonitor.queryCreatedEvent(
+                    new BasicQueryInfo(createQueryInfo(
+                            session,
+                            sql,
+                            PLANNING,
+                            Optional.empty(),
+                            sparkQueueName,
+                            Optional.empty(),
+                            queryStateTimer,
+                            Optional.empty(),
+                            warningCollector)));
+
+            TransactionId transactionId = transactionManager.beginTransaction(true);
+            session = session.beginTransactionId(transactionId, transactionManager, accessControl);
```

Comment:
Yes, `INVALID_SESSION_PROPERTY` error is thrown in this step. In presto classic, it should also have a query created event, otherwise, user error and query id is not tracked.

Reply:
Actually, we use QueryCompleteEvent in query lookup tool. The current behaviour of generating `QueryCreatedEvent` after session property validation is unchanged.